Create Spanish translation files

JPBarrio opened the pull request:
[This is my first entry in Github]
I have translated SettingsGuide into Spanish. 
Files can be uploaded once my proposed es_ES folder is accepted.
However, given the multiple translation possibilities for different 3D-world terms from English (i.e. 'ghosting', 'stringing', and so on), I seek collaboration, especially from the original Spanish translators in Cura to try standardizing spanish terminology.

Ghostkeeper commented:
`es_ES` is indeed accepted. It is the correct name for the directory. The mapping of language codes to user-readable text is [here](https://github.com/Ghostkeeper/SettingsGuide/blob/1f91403926b3565488926c45b05d43aafddaee4b/resources/qml/TranslationButton.qml#L50). It's using the [ISO 3166-1](https://en.wikipedia.org/wiki/ISO_3166-1) language code standard.

For the 3D industry terms, Cura's translation guide says to use whatever is most common in the industry. So if you believe that Spanish users tend to use the English words, keep them English, and if they use a Spanish term you'd use that. And it doesn't have to be perfect, so go by intuition I'd say.
I can't really help with the translations themselves. Just be sure to read the translation guidelines: https://github.com/Ghostkeeper/SettingsGuide/blob/master/contributing.md#translating

JPBarrio commented:
Thank you!
I have committed all translated files into the es_ES folder of the
SettingsGuide repository. Please excuse me as I am a newbie to git and this
is my first time after installing Git for Windows just to upload (commit)
them.
I can confirm the files are working in my latest Cura 5.0 installation.
As for the wording in Spanish, I have tried to follow as closely as
possible the naming already existing in the spanish version of Cura.
Sometimes I have included the English term within quotes (specially at
title texts) in order to help the reader clarify doubts.

El dom, 10 jul 2022 a las 20:04, Ghostkeeper ***@***.***>)
escribió:

> es_ES is indeed accepted. It is the correct name for the directory. The
> mapping of language codes to user-readable text is here
> <https://github.com/Ghostkeeper/SettingsGuide/blob/1f91403926b3565488926c45b05d43aafddaee4b/resources/qml/TranslationButton.qml#L50>.
> It's using the ISO 3166-1 <https://en.wikipedia.org/wiki/ISO_3166-1>
> language code standard.
>
> For the 3D industry terms, Cura's translation guide says to use whatever
> is most common in the industry. So if you believe that Spanish users tend
> to use the English words, keep them English, and if they use a Spanish term
> you'd use that. And it doesn't have to be perfect, so go by intuition I'd
> say.
> I can't really help with the translations themselves. Just be sure to read
> the translation guidelines:
> https://github.com/Ghostkeeper/SettingsGuide/blob/master/contributing.md#translating
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Ghostkeeper/SettingsGuide/pull/91#issuecomment-1179773010>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZYV5P6MTIFNRXMGQKDK6J3VTMGC3ANCNFSM52IOXI5A>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

JPBarrio commented:
Hello,
This is to ask for help!
I have committed all Spanish translation files from my local git
installation. The git bash messages are as follows:
------
*$ git status*




*On branch masterYour branch is ahead of 'origin/master' by 1 commit.  (use
"git push" to publish your local commits)nothing to commit, working tree
clean*
-------
When I try "git push", the following appears:
-------
*$ git push*
*remote: Permission to Ghostkeeper/SettingsGuide.git denied to JPBarrio.*
fatal: unable to access 'https://github.com/Ghostkeeper/SettingsGuide/':
The requested URL returned error: 403
-------
Is there something I am missing?



El dom, 10 jul 2022 a las 22:52, Juan Pablo Barrio Lera ***@***.***>)
escribió:

> Thank you!
> I have committed all translated files into the es_ES folder of the
> SettingsGuide repository. Please excuse me as I am a newbie to git and this
> is my first time after installing Git for Windows just to upload (commit)
> them.
> I can confirm the files are working in my latest Cura 5.0 installation.
> As for the wording in Spanish, I have tried to follow as closely as
> possible the naming already existing in the spanish version of Cura.
> Sometimes I have included the English term within quotes (specially at
> title texts) in order to help the reader clarify doubts.
>
> El dom, 10 jul 2022 a las 20:04, Ghostkeeper ***@***.***>)
> escribió:
>
>> es_ES is indeed accepted. It is the correct name for the directory. The
>> mapping of language codes to user-readable text is here
>> <https://github.com/Ghostkeeper/SettingsGuide/blob/1f91403926b3565488926c45b05d43aafddaee4b/resources/qml/TranslationButton.qml#L50>.
>> It's using the ISO 3166-1 <https://en.wikipedia.org/wiki/ISO_3166-1>
>> language code standard.
>>
>> For the 3D industry terms, Cura's translation guide says to use whatever
>> is most common in the industry. So if you believe that Spanish users tend
>> to use the English words, keep them English, and if they use a Spanish term
>> you'd use that. And it doesn't have to be perfect, so go by intuition I'd
>> say.
>> I can't really help with the translations themselves. Just be sure to
>> read the translation guidelines:
>> https://github.com/Ghostkeeper/SettingsGuide/blob/master/contributing.md#translating
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Ghostkeeper/SettingsGuide/pull/91#issuecomment-1179773010>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AZYV5P6MTIFNRXMGQKDK6J3VTMGC3ANCNFSM52IOXI5A>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>

Ghostkeeper commented:
Hi! Sorry for the delay again.

What you tried to do there is a normal way of working if you are a developer on a Github project. However Github doesn't allow you to directly change my source code. The proper way to contribute to someone else's project is as follows:

1. You need to fork the project to your own account. This creates a copy of the Settings Guide in your own account. It seems you already did this back in June, because [I found this fork in your profile](https://github.com/JPBarrio/SettingsGuide). Because that's your own copy, you'll have access rights to change that one.
2. You need to download that fork, not my original, to your computer. In Git bash you can do that by entering `git clone https://github.com/JPBarrio/SettingsGuide`. Previously you might've cloned `github.com/Ghostkeeper/SettingsGuide`.
3. Since you already have a pull request (this one) you should make sure you're on the branch of that pull request, that you called `patch-1`. You can do this by entering `git checkout patch-1` in your Git bash.
4. You then make changes to the source code (copy those files into your newly cloned fork), and make a commit. You already did this correctly, but you'll need to repeat that.
5. You can then push as you did, using `git push`. It will succeed this time, and you should see your changes appear [in your fork](https://github.com/JPBarrio/SettingsGuide) as well as [in your pull request](https://github.com/Ghostkeeper/SettingsGuide/pull/91).
6. Normally you'll then need to go to Github.com and create a pull request there. But you already did that, so that is unnecessary now.

I will then be able to review your changes.

For a guide from start to finish, you can use [this article](https://www.digitalocean.com/community/tutorials/how-to-create-a-pull-request-on-github). However since you already did a number of those steps, perhaps the steps I outlined above might be easier for you specifically.

Don't be afraid to make mistakes! And to ask for help if you need it.

Ghostkeeper commented:
It seems that these translations are machine-generated. I'm not allowing machine translations to be contributed here. Please see the [translation guidelines](https://github.com/Ghostkeeper/SettingsGuide/blob/master/contributing.md#translating).

JPBarrio commented:
I am using machine-generated translation, of course, but only after
checking it as thoroughly as possible. In all my professional life since
the eighties I've been struggling to make it easy translating from English.
I've seen lots of software doing strange, laughable sometimes,
translations, but always the balance is that it is worth
machine-translating first, then revising, rather than doing the hard work
by hand from the beginning. DeepL is the best translator I've encountered
so far.
This is all I can tell you. If you consider these texts do not deserve
being there, it is your decision. I myself will continue polishing it as
far as I can, as I am using Cura very frequently and SettingsGuide is a
very valuable tool to help polishing profiles.
All the best,
JP

El mar, 16 ago 2022 a las 12:47, Ghostkeeper ***@***.***>)
escribió:

> It seems that these translations are machine-generated. I'm not allowing
> machine translations to be contributed here. Please see the translation
> guidelines
> <https://github.com/Ghostkeeper/SettingsGuide/blob/master/contributing.md#translating>
> .
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Ghostkeeper/SettingsGuide/pull/91#issuecomment-1216470220>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZYV5P6SJXLIY3CTHR73WDLVZNWSTANCNFSM52IOXI5A>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

JPBarrio commented:
Thanks a lot for your guidance!

I think I'm starting to grasp this system, but for me in my sixties it is
still something complicated.
I have succeeded in copying the Spanish files into my patch-1 branch. I had
to do it in batches of less than 100 files at a time from github, as the
git bash gives strange messages:

The more recent changes involve the .svg images.
The "*git status*" command gives this output:




*On branch patch-1Your branch and 'origin/patch-1' have diverged,and have 2
and 34 different commits each, respectively.  (use "git pull" to merge the
remote branch into yours)*

*Changes to be committed:  (use "git restore --staged <file>..." to
unstage)        modified:   resources/translations/es_ES/images/[...].svg*
*        modified:  ...*

and my Git bash complains that the "*git pull*" operation would overwrite
some of my local files (that .svg files). It probably has something to do
with the automatic translation of LF to LF+CR (I edited these images using
notepad): After "*git add .*" I received a number of *"warning: in the
working copy of 'resources/translations/es_ES/images/[...].svg', LF will be
replaced by CRLF the next time Git touches it*".

The above happened after issuing a "*git push*" command with the following
output:





*To https://github.com/JPBarrio/SettingsGuide
<https://github.com/JPBarrio/SettingsGuide> ! [rejected]          patch-1
-> patch-1 (non-fast-forward)error: failed to push some refs to
'https://github.com/JPBarrio/SettingsGuide
<https://github.com/JPBarrio/SettingsGuide>'hint: Updates were rejected
because the tip of your current branch is behindhint: its remote
counterpart. Integrate the remote changes (e.g.hint: 'git pull ...') before
pushing again.*

Finally, the command

git commit --amend -m "svg images"

did the trick (I am using Emeditor as git text editor, and all commits use
it to create the "commit message" (or something like that) file that is
mandatory for the commit operation to be actually done, but this file is
always commented out so the commit is aborted; that's why I put the ..amend
modifier).
No more "modified: files" notifications, but there is still the "*git
status*" output about *"Your branch and 'origin/patch-1' have diverged, and
have 2 and 34 different commits each, respectively."*

Please have a look, I'll be happy to follow your advice once more!



El mié, 20 jul 2022 a las 23:07, Ghostkeeper ***@***.***>)
escribió:

> Hi! Sorry for the delay again.
>
> What you tried to do there is a normal way of working if you are a
> developer on a Github project. However Github doesn't allow you to directly
> change my source code. The proper way to contribute to someone else's
> project is as follows:
>
>    1. You need to fork the project to your own account. This creates a
>    copy of the Settings Guide in your own account. It seems you already did
>    this back in June, because I found this fork in your profile
>    <https://github.com/JPBarrio/SettingsGuide>. Because that's your own
>    copy, you'll have access rights to change that one.
>    2. You need to download that fork, not my original, to your computer.
>    In Git bash you can do that by entering git clone
>    https://github.com/JPBarrio/SettingsGuide. Previously you might've
>    cloned github.com/Ghostkeeper/SettingsGuide.
>    3. Since you already have a pull request (this one) you should make
>    sure you're on the branch of that pull request, that you called patch-1.
>    You can do this by entering git checkout patch-1 in your Git bash.
>    4. You then make changes to the source code (copy those files into
>    your newly cloned fork), and make a commit. You already did this correctly,
>    but you'll need to repeat that.
>    5. You can then push as you did, using git push. It will succeed this
>    time, and you should see your changes appear in your fork
>    <https://github.com/JPBarrio/SettingsGuide> as well as in your pull
>    request <https://github.com/Ghostkeeper/SettingsGuide/pull/91>.
>    6. Normally you'll then need to go to Github.com and create a pull
>    request there. But you already did that, so that is unnecessary now.
>
> I will then be able to review your changes.
>
> For a guide from start to finish, you can use this article
> <https://www.digitalocean.com/community/tutorials/how-to-create-a-pull-request-on-github>.
> However since you already did a number of those steps, perhaps the steps I
> outlined above might be easier for you specifically.
>
> Don't be afraid to make mistakes! And to ask for help if you need it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Ghostkeeper/SettingsGuide/pull/91#issuecomment-1190758846>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZYV5P5BOTJZ2VTWAIZJ27DVVBTCJANCNFSM52IOXI5A>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
